streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -74,16 +74,16 @@
      * Create a new {@code KTable} that consists of all records of this {@code KTable} which satisfy the given
      * predicate.
      * All records that do not satisfy the predicate are dropped.
-     * For each {@code KTable} update the filter is evaluated on the update record to produce an update record for the
+     * For each {@code KTable} update the filter is evaluated on the updated record to produce an updated record for the
```

Comment:
This was actually intentionally -- it the record that does the update , ie, the "update record" (the new record), not the record that got updated (ie, the old record).

\cc @joel-hamill WDYT?

Reply:
I agree with @mjsax IMHO it should be the `update record` my interpretation is the latest record for a given key.

Reply:
What do others think about For each update record in the {@code KTable}, a filter is evaluated to produce an updated {@code KTable} containing only those update records satisfying the predicate

Reply:
Sounds good to me.

I am also fine in changing from `update` to `updated` -- hope @joel-hamill can give some advice what reads better.

Reply:
To me this would make more sense: 
`     * For each {@code KTable} update, the filter is evaluated based on the current update record and then an update record is produced for the result {@code KTable}.`